src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -261,15 +262,25 @@ public void write() {
               }
             } else {
               SinkRecord projectedRecord = SchemaUtils.project(record, currentSchema, compatibility);
-              writeRecord(projectedRecord);
-              buffer.poll();
-              if (shouldRotate(now)) {
+              String partition = partitioner.encodePartition(projectedRecord);
+              if (currentPartition != null && !currentPartition.equals(partition)) {
+                currentPartition = partition;
```

Comment:
@ewencp what about cron expression? The problem with `rotate.schedule.ms` is values that does not fit well. Like 126764ms.